### PR TITLE
Added Time Profiler

### LINKED ISSUE #589: Basic Time Profiling into contrib
Following https://github.com/pytorch/ignite/issues/63#issuecomment-523366321
we can provide some basic time profiling. 

cc @willprice what do you think about this profiler, any code, API updates to suggest ?

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #589

Description: Added basic time profiling handler

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

@Patil2099, do not remove above text, please!

### CODE REVIEW

**Comment by vfdev-5** [comment]
@Patil2099 please keep PR's template.

- CI tells that you need ti fix flake8

Sorry, I think we need to rename the file `time_profiler.py` to `time_profilers.py`

**Comment by vfdev-5** [comment]
@Patil2099 do you plan to work on this ?

**Comment by Patil2099** [comment]
I will finish this by today @vfdev-5

**Comment by Patil2099** [comment]
@vfdev-5 Is this Correct?

**Comment by vfdev-5** [comment]
@Patil2099, now you need to write some tests to check newly introduced class.

PS: I restored PR template text in the first message, please do not remove it.

File: ignite/contrib/handlers/time_profilers.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
+from collections import OrderedDict
+
+import torch
+
+from ignite.engine import Engine, Events
+from ignite.handlers import Timer
+
+
+class BasicTimeProfiler(object):
+
```
Comment (vfdev-5) [inline ignite/contrib/handlers/time_profilers.py]: Here we need to add docstring. For example,https://github.com/pytorch/ignite/blob/4253bfbf40011288ed952e73344e8d470e164b9f/ignite/handlers/early_stopping.py#L7

**Comment by vfdev-5** [comment]
@Patil2099 any updates ?

**Comment by Patil2099** [comment]
@vfdev-5 Can you help with the testcases?
Thanks!

**Comment by vfdev-5** [comment]
@Patil2099 idea is to cover the code with tests. Please, first read https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#contributing-to-ignite and then take a look at other tests done in ignite. Test cases for this class could be time measurements on a synthetic engine with handlers and update function that uses `time.sleep`.
Please, ask more concrete questions.

**Comment by vfdev-5** [comment]
close the PR as stale

**Pull request closed without merging**